dynawo/sources/Modeler/DataInterface/DYNCriteria.h

```diff
@@ -44,10 +44,18 @@ class Criteria {
    *
    * @return true if the criteria is respected, false otherwise
    */
-  virtual bool checkCriteria(bool finalStep) const = 0;
+  virtual bool checkCriteria(bool finalStep) = 0;
+
+  /**
+   * @brief returns the list of failing criteria ids
+   *
+   * @return list of failing criteria ids
+   */
+  const std::vector<std::string>& getFailingCriteriaIds() const {return failingCriteria_;}
 
```

Comment:
If I get it right, what we retrieve is the log message and not really the id of the criterion, isn't it?

Reply:
Yes this is correct. To display which criterion failed it is clearer with the full message. I guess you want me to change the method name?

Reply:
Right, it would be nice to adapt the method name and comments, sorry for not having been accurate enough in the comment!